src/RamOperation.h

```diff
@@ -865,9 +865,50 @@ class RamAggregate : public RamRelationOperation, public RamAbstractAggregate {
     }
 };
 
+/**
+ * @class RamParallelAggregate
+ * @brief Parallel Aggregation function applied on some relation
+ *
+ * For example:
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * PARALLEL t0.0 = COUNT FOR ALL t0 IN A
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * Applies the function PARALLEL COUNT to determine the number
+ * of elements in A.
+ */
+class RamParallelAggregate : public RamAggregate, public RamAbstractParallel {
+public:
+    RamParallelAggregate(std::unique_ptr<RamOperation> nested, AggregateOp fun,
+            std::unique_ptr<RamRelationReference> relRef, std::unique_ptr<RamExpression> expression,
+            std::unique_ptr<RamCondition> condition, int ident)
+            : RamAggregate(std::move(nested), fun, std::move(relRef), std::move(expression),
+                      std::move(condition), ident) {}
+
+    RamParallelAggregate* clone() const override {
+        return new RamParallelAggregate(souffle::clone(&getOperation()), function,
+                souffle::clone(relationRef), souffle::clone(expression), souffle::clone(condition),
+                identifier);
+    }
+
+protected:
+    void print(std::ostream& os, int tabpos) const override {
+        os << times(" ", tabpos);
+        os << "PARALLEL t" << getTupleId() << ".0=";
+        RamAbstractAggregate::print(os, tabpos);
+        os << "FOR ALL t" << getTupleId() << " ∈ " << getRelation().getName();
+        if (!isRamTrue(condition.get())) {
+            os << " WHERE " << getCondition();
+        }
+        os << std::endl;
+        RamRelationOperation::print(os, tabpos + 1);
+    }
+};
+
 /**
  * @class RamIndexAggregate
  * @brief Indexed aggregation on a relation
+ *
+ * TODO (rachel): Fix description!! I'm not sure what this represents
```

Comment:
This would be a good time to fix the description. If you want to expand on the brief version, then something about potentially using an index to find a smaller range for the aggregation?